docs/devs/plugin-migration-to-alpha-14.md

```diff
@@ -0,0 +1,523 @@
+# Migrating Plugins to Alpha 14
+
+Between version 2.0.0-alpha.13 and 2.0.0-alpha.14 the codebase underwent major refactoring and was ported to TypeScript.
+In order to support alpha 14 there are several major changes that need to be done to existing plugins, these are in no particular order
+
+
+## Rename master and slave
+
+All references to slave and master in the code was renamed to controller and host respectively which means that code referencing these in plugins must be renamed.
+In short this means performing a global search and replace on the codebase replacing:
+
+- `master` with `controller`
+- `Master` with `Controller`
+- `slave` with `host`
+- `Slave` with `Host`
+
+in all places it occurs, even as part of longer names such as changing `BaseMasterPlugin` to `BaseControllerPlugin`.
+
+
+## Change lib imports to import directly from the top level package
+
+Previously code could import sub packages of lib such as lib/lua_tools.
+These imports need to be changed to all import from lib directly.
+
+**Alpha 13**
+```js
+const libLuaTools = require("@clusterio/lib/lua_tools");
+const libErrors = require("@clusterio/lib/errors");
+
+// Using imports
+new libLink.RequestError(...)
+libLuaTools.escapeString(...)
+```
+
+**Alpha 14 (JS)**
+```js
+const lib = require("@clusterio/lib");
+
+// Using library components
+new lib.RequestError(...)
+lib.escapeString(...)
+```
+
+**Alpha 14 (TS)**
+```ts
+import * as lib from "@clusterio/lib";
+
+// Using library components
+new lib.RequestError(...)
+lib.escapeString(...)
+```
+
+
+## Change import of base plugins
+
+Previously `BaseMasterPlugin` and `BaseInstancePlugin` were imported from `@clusterio/lib`, these have been moved to `@clusterio/controller` and `@clusterio/host` respectively.
+
+**Alpha 13**
+```js
+const { BaseMasterPlugin } = require("@clusterio/lib");
+const { BaseInstancePlugin } = require("@clusterio/lib");
+```
+
+**Alpha 14 (JS)**
+```js
+const { BaseControllerPlugin } = require("@clusterio/controller");
+const { BaseInstancePlugin } = require("@clusterio/host");
+```
+
+**Alpha 14 (TS)**
+```ts
+import { BaseControllerPlugin } from "@clusterio/controller";
+import { BaseInstancePlugin } from "@clusterio/host";
+```
+
+
+## Change main entrypoint of plugin
+
+Previously the info of a plugin would be loaded by requiring `./plugin_dir/info`, this has been changed to require `./plugin_dir` which in turn means that `info.js` is no longer the default entrypoint.
+To solve this you can either add `"main": "info.js",` to package.json or rename `info.js` to `index.js` (the rest of this guide assumes the entrypoint was renamed to `index.js`).
+The motivation behind this change was to allow TypeScript to output its build to a sub-folder of the plugin, if you're using TypeScript you'll likely have the entrypoint be located in `dist/index.js` instead.
+
+The expected place the declaration of the plugin is exported also changed from `module.exports` to `module.exports.plugin`:
```

Comment:
I shouldn't be one to talk about grammar, because mine is arwful, but...
```suggestion
The expected place where the declaration of a plugin is exported has also changed from `module.exports` to `module.exports.plugin`:
```